basesandbox/src/execution/executee.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::ipc::*;
+
+// Interface for the sandboxee written in Rust
+pub struct Context<T: Ipc> {
+    /// ipc will be given with Some, but module may take it
+    /// However, the ipc must be return back here before the module terminates
+    pub ipc: Option<T>,
```

Comment:
If the executee want to utilize given ipc with its own purpose, it will `take()` it. However the ipc has to be used in destruction phase, the excutee must return back it.

Reply:
This is because I assumed `Ipc` can be not `Sync`

Reply:
It seems awkward. Let's talk again when actually the thread takes the ipc field.